Assets/Scripts/Augment/AugmentImplementations/ContinuousDamage.cs

```diff
@@ -0,0 +1,40 @@
+using UnityEngine;
+
+public class ContinuousDamage : MonoBehaviour
+{
+    [SerializeField]
+    private float damage;
+
+    [SerializeField]
+    private float damageRate;
+
+    private float lastDamageTime = -100;
+
+    [HideInInspector]
+    public PlayerManager source;
+
+    private HitboxController hitbox;
+
+    void Awake()
+    {
+        hitbox = transform.parent.GetComponent<HitboxController>();
+        if (hitbox && hitbox.health)
+        {
+            hitbox.health.onDeath += OnDeath;
+        }
+    }
+
+    private void OnDeath(HealthController healthController, float damage, DamageInfo info)
+    {
+        this.enabled = false;
+    }
+
+    private void Update()
+    {
+        if (hitbox && Time.time - lastDamageTime > 1 / damageRate)
+        {
```

Comment:
We might want to add a onHit invocation here, but it's not strictly necessary yet

Reply:
That's a little outside the aim of this component, but it seems like the sort of thing we would want for our _actual_ hat barrel implementation